Simplenote/src/main/java/com/automattic/simplenote/utils/ReviewAccountVerifier.java

```diff
@@ -0,0 +1,74 @@
+package com.automattic.simplenote.utils;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.automattic.simplenote.Simplenote;
+import com.automattic.simplenote.models.Account;
+import com.simperium.client.Bucket;
+import com.simperium.client.BucketObjectMissingException;
+
+import static com.automattic.simplenote.models.Account.KEY_EMAIL_VERIFICATION;
+
+/**
+ * When a user login into their account, Simplenote starts indexing the different buckets including Bucket<Account>.
+ * We setup a OnNetworkChangeListener on the account bucket to listen when the bucket is indexed. One of the parameters
+ * of the callback onNetworkChange called is Bucket.ChangeType. If this parameter is INDEX, then we know that the
+ * account bucket was indexed (synchronized). When the account bucket is synchronized, we can check whether the account
+ * was verified or not
+ */
+public class ReviewAccountVerifier implements Bucket.OnNetworkChangeListener<Account> {
+    private enum Status {
+        UNKNOWN,
+        UNVERIFIED,
+        VERIFIED,
+    }
+
+    private final Simplenote simplenote;
+    private Status currentStatus = Status.UNKNOWN;
+
+    public ReviewAccountVerifier(Simplenote simplenote) {
+        this.simplenote = simplenote;
+    }
+
+    public void clearStatus() {
+        currentStatus = Status.UNKNOWN;
+    }
+
+    @Override
+    public void onNetworkChange(Bucket<Account> bucket, Bucket.ChangeType type, String key) {
+        // If we already know the answer, we avoid computing it again
```

Comment:
for boot this would be good enough but we probably also want to monitor the bucket for changes. it's definitely possible to de-verify an account so we might get change events of `MODIFY` or `REMOVE` that would update the verification status.

it's also a trivial computation so there's no need to avoid it.

Reply:
Question: if we get a `REMOVE` event, does the account will be available if we try to get it: 
```java
Account account = bucket.get(KEY_EMAIL_VERIFICATION);
```

Reply:
> does the account will be available if we try to get it with:

I think you're asking if the verification entity will still be available in the bucket if there's a `REMOVE` event. if we _do_ get that event then we'd still have to check. if `key` is the same as `KEY_EMAIL_VERIFICATION` then no, we wouldn't expect the property to exist and we would immediately know that we are now `UNVERIFIED`. if it's any other value then we can ignore the event because it was some other entity that was removed.

Reply:
> if key is the same as KEY_EMAIL_VERIFICATION then no, we wouldn't expect the property to exist and we would immediately know that we are now UNVERIFIED

That defines the logic for this event type. Thanks!